webapp/channels/src/components/user_settings_modal/advanced_display_settings/utils.tsx

```diff
@@ -0,0 +1,307 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {defineMessage, FormattedMessage} from 'react-intl';
+
+import type {FieldsetCheckbox} from 'components/widgets/modals/components/checkbox_setting_item';
+import type {FieldsetRadio} from 'components/widgets/modals/components/radio_setting_item';
+
+import {isMac} from 'utils/user_agent';
+
+export enum AdvanceSettings {
+    CTRL_SEND='ctrlSend',
+    SEND_ON_CTRL_ENTER= 'send_on_ctrl_enter',
+    CODE_BLOCK_CTRL_ENTER= 'code_block_ctrl_enter',
+    FORMATTING='formatting',
+    JOIN_LEAVE='join_leave',
+    NAME_DISABLE_CLIENT_PLUGINS= 'disable_client_plugins',
+    NAME_DISABLE_TELEMETRY= 'disable_telemetry',
+    NAME_DISABLE_TYPING_MESSAGES= 'disable_typing_messages',
+    UNREAD_SCROLL_POSITION= 'unread_scroll_position',
+    SYNCED_DRAFTS_ARE_ALLOWED = 'synced_drafts_are_allowed'
+}
+
+export const enum OnAndOff {
```

Comment:
I noticed that we seem to store these values in preferences as `true`/`false` instead of `on`/`off` and that there seems to be a lot of logic to translate back and forth between those like we use for `formattingSectionContent` in the previous file. Would it be easier instead to keep these values as `true`/`false` when passing them to the controls to avoid having to convert them back and forth repeatedly?